PWGMM/Rivet/DataModel/RivetAOs.cxx

```diff
@@ -0,0 +1,232 @@
+// Copyright 2023-2099 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+//
+#include "RivetAOs.h"
+#include <YODA/IO.h>
+#include <TSystem.h>
+#include <TClass.h>
+#include <vector>
+#include <string>
+#include <Rivet/AnalysisHandler.hh>
+#include <Rivet/Tools/RivetPaths.hh>
+// Sigh - linter needs additional headers already included
+
+namespace o2
+{
+namespace rivet
+{
+//----------------------------------------------------------------
+RivetAOs::RivetAOs()
+  : mData("")
+{
+  mLoadPaths.SetOwner(true);
+}
+//----------------------------------------------------------------
+RivetAOs::RivetAOs(const RivetAOs& o)
+  : mData(o.mData),
+    mEquivalent(o.mEquivalent)
+{
+  mLoadPaths.SetOwner(true);
+}
+//----------------------------------------------------------------
+void RivetAOs::AddLoadPath(const char* path)
+{
+  mLoadPaths.Add(new TObjString(path));
+}
+//----------------------------------------------------------------
+void RivetAOs::ReadIn(std::istream& in)
+{
+  mData.String() = std::string((std::istreambuf_iterator<char>(in)),
+                               std::istreambuf_iterator<char>());
+}
+//----------------------------------------------------------------
+void RivetAOs::WriteOut(std::ostream& out) const
+{
+  out << mData.GetString() << std::flush;
+}
+//----------------------------------------------------------------
+void RivetAOs::SaveAs(const char* filename, Option_t*) const
+{
+  std::ofstream out(filename);
+  WriteOut(out);
+}
+//----------------------------------------------------------------
+void RivetAOs::Print(Option_t*) const
+{
+  std::cout << "=== Rivet Analysis Objects:\n";
+  WriteOut(std::cout);
+  std::cout << "=== " << std::endl;
+}
+#if 0
+//----------------------------------------------------------------
+void RivetAOs::merge(o2::mergers::MergeInterface* const other)
+{
+  Info("merge", "O2 merge interface member function called");
+  RivetAOs* o = dynamic_cast<RivetAOs*>(other);
+  if (!o) { // Sigh - annoying linter prefers `!` over `not`
```

Comment:
Yeah, but it could you to see the thing.  I would argue that the code is _much_ easier to read with `not o` than `!o`.  In the latter case, it is _very_ easy to overlook the `!` - especially if the negated condition is long or involves `(...)`.   Please consider to remove that linter rule.